### ISSUE TITLE
[Feature Request] option to only clear terminal if in watch mode

### ISSUE THREAD

**Issue by andykais** [danillouz/clean-terminal-webpack-plugin#4]
According to https://github.com/webpack/webpack/issues/3460, we can detect when the compiler is in watch mode. I want an option to this plugin like `onlyInWatchMode: boolean` which will only clear the screen when `webpack --watch` is enabled.

I am happy to open a PR for this if it would be accepted!

**Comment by andykais** [comment]
possible improvement: change the options to look like so:
this ignores using `NODE_ENV` (which currently is checked in the plugin), but I think it makes more sense to only use variables pertaining to webpack. Let me know your thoughts!

```javascript
// values shown would be defaults
new CleanTerminalWebpackPlugin({
  always: false,
  inWatchMode: true, // webpack --watch
  inDevelopmentMode: false, // webpack --mode=development
  inProductionMode: false, // webpack --mode=production
})

**Comment by danillouz** [comment]
Hi Andrew! I'm happy to merge the PR if this feature would be helpful to you.

I'm curious though, what your use case is to _not_ use it in watch mode, while developing. I haven't used this plugin nor webpack for a long time, but when I did, it was only while watching for changes during development. At the time it was also considered "good practice" to have separate webpack configs for dev and prod.

I think your suggestion makes sense to make the `onlyInWatchMode` option configurable. As for the other options I don't really get what the added value might be of for example `inProductionMode` and others, for the reasons stated above.

**Comment by danillouz** [comment]
Published as [v2.0.0](https://www.npmjs.com/package/clean-terminal-webpack-plugin)
